CMake/External_Protobuf.cmake

```diff
@@ -1,43 +1,50 @@
-if (WIN32)
-  # Build option for windows not yet generated
-  message( FATAL_ERROR "Protobuf on windows not yet supported" )
-endif()
 
-# Check that python and protobuf versions are compatible
-if(fletch_BUILD_WITH_PYTHON AND fletch_ENABLE_Protobuf)
-  # Note the python protobuf wrapper is not installed here.
-  # Instead it must be installed via `pip install protobuf`
-  if (${Protobuf_version} LESS 3.0 AND fletch_PYTHON_MAJOR_VERSION MATCHES "^3.*")
-    message(ERROR " Must use Protobuf >= 3.x with Python 3.x")
+if (Protobuf_SELECT_VERSION STREQUAL "3.9.0")
+  ExternalProject_Add(Protobuf
+#    PREFIX protobuf
+    URL ${Protobuf_url}
+    URL_MD5 ${Protobuf_md5}
+    SOURCE_SUBDIR ./cmake
+    ${COMMON_EP_ARGS}
+    ${COMMON_CMAKE_EP_ARGS}
+#    UPDATE_COMMAND
+#        COMMAND ${CMAKE_COMMAND} -DPULSE_IL2CPP_PATCH=${PULSE_IL2CPP_PATCH} -Dprotobuf_source=${protobuf_SRC} -Dprotobuf_patch=${protobuf_Patch} -P ${protobuf_Patch}/Patch.cmake
+    CMAKE_ARGS
+       ${COMMON_CMAKE_ARGS}
+       -Dprotobuf_BUILD_TESTS:BOOL=OFF
+       -Dprotobuf_BUILD_EXAMPLES:BOOL=OFF
+       -Dprotobuf_BUILD_SHARED_LIBS:BOOL=${BUILD_SHARED_LIBS}
```

Comment:
Please read this tid bit about protobuf linkage
https://github.com/protocolbuffers/protobuf/tree/master/cmake#dlls-vs-static-linking

I suggest we always build protobuf as static libs, and they should only be linked into the protobuf arrow and nothing else. Should work great in the kwiver architecture.